src/main/java/htsjdk/samtools/fastq/FastqEncoder.java

```diff
@@ -106,6 +119,9 @@ public static SAMRecord asSAMRecord(final FastqRecord record, final SAMFileHeade
         // set the basic information from the FastqRecord
         samRecord.setReadName(readName);
         samRecord.setReadBases(record.getReadBases());
+        if (qualityHeaderToComment) {
+            samRecord.setAttribute(SAMTag.CO.name(), record.getBaseQualityHeader());
```

Comment:
make sure that tabs are not part of this string....as you'll destroy the bam otherwise. 
(please test this)

Reply:
Actually, this is a difficult issue: in bwa, for example, the quality-header could be use to assign tags to SAM/BAM. From http://bio-bwa.sourceforge.net/bwa.shtml:

```
-C | Append append FASTA/Q comment to SAM output. This option can be used to transfer read meta information (e.g. barcode) to the SAM output. Note that the FASTA/Q comment (the string after a space in the header line) must conform the SAM spec (e.g. BC:Z:CGTAC). Malformated comments lead to incorrect SAM output.
```

Maybe the way to go here is to parse the quality header for tag-formatted comments and if present add to the attributes; otherwise, add to the `SAMTag.CO` by converting any tab to spaces. What do you think?

Reply:
I think that if there's some standard, (e.g. the one in bwa) we should use it. I don't like the idea of "either/or"....can you just use SAM-style tags and not allow for free text?

Reply:
I don't think that the `bwa` quality header is a standard, but an ad-hoc solution to allow including per-read information from FASTQ input reads into the output of `bwa`. I still would like to allow for free-text, because in the wild there are FASTQ files with that kind of header, and there are options when downloading from SRA/ENA servers to include free-text information in quality headers (e.g., sratoolkit's `fastq-dump` command allows custom `--defline-qual`, which won't complies to SAM-style tags).

Reply:
In that case, I would advise for a parameter switch for the different modes, rather than something that depends on "if present".....

Reply:
I added a even better configuration mode: use a `BiConsumer` to add extra information from the `FastqRecord` to the `SAMRecord` is a custom way: I included an implementation for the `CO` tag (converting tabs into spaces) and we can also include the one for setting the tags if you feel that is necessary.

Reply:
I did add also another custom encoder.